### ISSUE TITLE
Go to File Search Should Not Prioritize Alphabetical "Fuzzy" Match Over Exact Match in Results

### ISSUE THREAD

**Issue by TheCookieLab** [apache/netbeans#5439]
### Description

When using the Go to File search the results appear to be sorted alphabetically which means the default selection is commonly not what the user is looking for.

For example:
In the below screenshot I search for "DirectionalMovementOfIndicatorTest" and in the results, the default selection is a "fuzzy" match despite the exact match was also found.  

![Search Annoyance](https://user-images.githubusercontent.com/6332181/216838893-75dd713c-db9a-4ac5-a7bf-d0477ff2c45f.PNG)

It would be nice if there's an exact match it's placed first and selected by default. The fuzzy matches can then come after, sorted via the existing criteria.

### How to reproduce

1. In a new or existing project create three new class files, all named similarly. For example:

DirectionalMovementDownOfIndicatorTest.java
DirectionalMovementOfIndicatorTest.java
DirectionalMovementUpOfIndicatorTest.java

2. Open the "Go to File" search (alt+shift+o on Windows)
3. Search for a file which name is not the first in natural sorting order (i.e. DirectionalMovementOfIndicatorTest.java)
4. Press ENTER as soon as the search results pop up cause you're in the zone
5. Notice with chagrin that you are not where you expected to be. Level of confusion correlated with how similar the classes are.

### Use case/motivation

This is low priority but if a UX uplift initiative ever gets going this should be included.

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by mbien** [comment]
added https://github.com/apache/netbeans/labels/Contribution%20welcome label. I suppose you didn't mean to select that you are planning on opening a PR for this?

**Comment by TheCookieLab** [comment]
@mbien good catch thanks. Yeah I'm probably not the best suited person for this.

**Comment by mbien** [comment]
@TheCookieLab didn't mean it like that. Contribution is always welcome!

**Comment by albilu** [comment]
There is a setting to change the ordering to match Similarity instead of alphabetic order. Try it and tell if better:
![image](https://user-images.githubusercontent.com/35330562/232288464-c182b2ea-5479-4ad7-abc8-a1f30c15794d.png)
